bin/all_sky_search/pycbc_rerank_dq

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python
+""" Merge and rerank clustered DQ data
+"""
+import logging, argparse, numpy, h5py
+import pycbc
+from os import path
+from pycbc.version import git_verbose_msg as version
+
+parser = argparse.ArgumentParser(description=__doc__)
+parser.add_argument('--version', action='version', version=version)
+parser.add_argument('--verbose', action="store_true",default=False)
+parser.add_argument('--no-low-dq-cutoff', action="store_true",
+                    help="Allow dq values less than 0. "
+                         "Otherwise values less than 0 are set to 0")
+parser.add_argument("--ifo", type=str,required=True,
+                     help="interferometer to analyze")
+parser.add_argument("--max-stat", type=float,default=8,
+                     help="maximum allowed dq value. "
+                           "Only used if --rate-file is not used. "
+                            "[default=8]")
+parser.add_argument("--input-file", required=True,nargs='+',
+                     help="List of files containing the dq timeseries")
+parser.add_argument("--rate-file", type = str, default = None,
+                     help="File containing the mapping from each "
+                           "dq bin to the trigger rate")
+parser.add_argument("--bank-file", type = str, default = None,
+                     help="File containing the template bank")
+parser.add_argument("--dq-type", type = str, default = 'dq',
+                     help="Name of dq type to be included in output file "
+                          "metadata [default='dq']")
+parser.add_argument("--output-file", required=True,
+                     help="name of output file")
+
+args = parser.parse_args()
+pycbc.init_logging(args.verbose)
+
+ifo = args.ifo
+
+log_like = numpy.array([])
+times = numpy.array([])
+for filename in args.input_file:
+    logging.info('Reading file %s...', filename)
+    with h5py.File(filename, 'r') as g:
+        g_log_like = g[ifo + '/dq_val'][:]
+        g_times = g[ifo + '/times'][:]
+    log_like = numpy.concatenate((log_like,g_log_like))
+    times = numpy.concatenate((times,g_times))
+
+log_like_ranking = numpy.argsort(log_like)
+len_log_like = len(log_like)
+
+times = times[log_like_ranking]
+log_like_vals = numpy.array(log_like[log_like_ranking])
+
+log_like_dict = {}
+percent_dict = {}
+
+if not args.rate_file or args.bank_file:
+    raise TypeError('Must supply either a bank or dq rates file')
```

Comment:
This check looks a bit strange. Is the idea that you need one of the two options, but not both? If so, try using a [mutually exclusive option group](https://docs.python.org/3/library/argparse.html#argparse.ArgumentParser.add_mutually_exclusive_group) in your argument parser. Then this explicit check should not be needed.

Reply:
Yes, basically if a DQ rate file is provided, then information from that file is used. Otherwise the bank file is used to apply an analytic model. I've implemented your suggestion